sched/sched/sched_note.c

```diff
@@ -370,6 +437,10 @@ void sched_note_start(FAR struct tcb_s *tcb)
   strncpy(note.nsa_name, tcb->name, CONFIG_TASK_NAME_SIZE + 1);
 
   length = SIZEOF_NOTE_START(namelen + 1);
+
+#ifdef NOTE_FEATURE_TASK_NAME
+  note_taskname_put(tcb->pid, tcb->name);
```

Comment:
If the note already contain task name and pid, why we need save pid and name additionally?

Reply:
Different to the existing sched_note feature, in task trace support, the period to collect the note (trace) data and to dump the trace result are completely separated.
So, when dumping the result, the task recorded in the trace data may not exist and its task name string may already freed.
It is the reason why task name and its pid should be saved.

Reply:
Sorry, my comment didn't answer to your question.
It is because only note_start_s has the task name information. We can observe the task name only for newly created tasks and cannot for existing tasks.
Adding task name member to note_suspend_s and note_resume_s solves it but it increases the trace data size considerably because the task name is recorded for each context switches.